src/main/java/com/faforever/api/avatar/AvatarService.java

```diff
@@ -47,12 +49,15 @@ public AvatarService(AvatarRepository avatarRepository, FafApiProperties propert
   public void createAvatar(AvatarMetadata avatarMetadata, String originalFilename, InputStream imageDataInputStream, long avatarImageFileSize) {
     final Avatar avatarToCreate = new Avatar();
     final String normalizedAvatarFileName = NameUtil.normalizeFileName(originalFilename);
-    String url = String.format(properties.getAvatar().getDownloadUrlFormat(), normalizedAvatarFileName);
-    avatarRepository.findOneByUrl(url).ifPresent(existingAvatar -> {
+    String url = String.format(properties.getAvatar().getBaseUrl(), normalizedAvatarFileName);
```

Comment:
I not sure if this should be here. This is only during creation, but we should compute this attribute at read time also. see comment below